packages/shared/src/types.ts

```diff
@@ -125,16 +125,25 @@ export enum ErrorCode {
 
 export type ResolutionReason = keyof typeof StandardResolutionReasons | (string & Record<never, never>);
 
+/**
+ * A structure which supports definition of arbitrary properties, with keys of type string, and values of type boolean, string, or number.
+ * 
+ * This structure is populated by a provider for use by an Application Author (via the Evaluation API) or an Application Integrator (via hooks).
+ */
+export type FlagMetadata = Record<string, string | number | boolean>;
```

Comment:
This complies with [requirement 2.2.9](https://openfeature.dev/specification/sections/providers#requirement-229) but doesn't have a test. Do you think one is needed? I couldn't find the other 2.x tests.

Reply:
Requirements that are structural (types) or prescriptions to providers don't really require tests here IMO.